src/utils/FormInputHandling.ts

```diff
@@ -15,16 +25,108 @@ async function getData(currentPage: number, limit: number) {
   };
 }
 
+// getting data from db when session_id is null
+async function getDataNoLinks(currentPage: number, limit: number) {
+  const offset = currentPage * limit;
+  const { data } = await instance.get("api/session", {
+    params: {
+      session_id_is_null: true,
+      offset,
+      limit: limit + 1,
+      sort_order: "desc",
+      platform: "quiz",
+    },
+  });
+  const hasMore = data.length > limit;
+  const items = hasMore ? data.slice(0, -1) : data;
+  return {
+    data: items,
+    hasMore,
+  };
+}
+
+function formatDateTime(date: string, time: string) {
+  const [year, month, day] = date.split("-").map(Number);
+  const [hour, minute] = time.split(":").map(Number);
+
+  const combinedDate = new Date(year, month - 1, day, hour, minute);
+
+  return combinedDate;
+}
 // post data to the server
 async function postFormData(formData: RowType) {
-  const currentDate = new Date().toLocaleDateString();
+  const { student, test, timeline, dateCreated, session } = formData;
 
-  const res = await instance.post("/quiz", {
-    dateCreated: currentDate,
-    ...formData,
-  });
+  let start_time = formatDateTime(
+    timeline.startDate as string,
+    timeline.startTime as string
+  );
+  let end_time = formatDateTime(
+    timeline.endDate as string,
+    timeline.endTime as string
+  );
+
+  const requestBody = {
+    name: test.name,
+    platform: test.platform,
+    platform_link: "",
+    portal_link: "", //responds to the portal link
+    start_time: start_time,
+    end_time: end_time,
+    is_active: "",
+    purpose: {
+      type: "attendance",
+      params: "quiz",
+    },
+    repeat_schedule: { type: "weekly", params: [1, 2, 3, 4, 5, 6, 7] }, // this is hardcoded needs to be changed
+    session_id: "",
+    platform_id: "",
+    // form_schema_id: session.form_schema_id, // should be signup_form_name
+    signup_form_name: "Haryana Registration Form",
+    type: "sign-in",
```

Comment:
Why is this hard coded?

Reply:
We were testing it, like the cmsID we are filling manually for now